### PR TITLE
Java: support more libraries in hardcoded-credentials queries

### PR DESCRIPTION
**Pull request by smowton**
This is abandoned external PR https://github.com/github/codeql/pull/6716, but I have also written tests and ported the hardcoded-creds queries to use inline-expectations tests so it's practically possible to verify that the results are as expected.

### CODE REVIEW

**Review by atorralba — APPROVED** [review]
LGTM! Thanks for doing this.

I always wonder, though — how much of this should be covered by secret scanning and not CodeQL?

File: java/ql/lib/semmle/code/java/security/HardcodedCredentialsComparison.qll [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Provides classes and predicates to detect comparing a parameter to a hard-coded credential.
+ */
+
```
Comment (atorralba) [inline java/ql/lib/semmle/code/java/security/HardcodedCredentialsComparison.qll:4]: Oh, and a little NIT: maybe we should have `import java` here for consistency. I guess the imports should be `private` too, but this doesn't seem like it's going to be used anywhere except in the query itself, so 🤷

Comment (smowton) [reply java/ql/lib/semmle/code/java/security/HardcodedCredentialsComparison.qll:4]: Added the imports. Re: who should cover this, my impression was the involvement of dataflow analysis makes it appropriate for CodeQL, but it's possible I'm underestimating what secret scanning can achieve.

Comment (atorralba) [reply java/ql/lib/semmle/code/java/security/HardcodedCredentialsComparison.qll:4]: Well, that makes sense actually! 👍

**Review by atorralba — APPROVED** [review]

**Pull request merged**